### PR TITLE
Improved bootstrap of Prism instance in test harness

### PR DESCRIPTION
**Pull request by StefanDywersant**
## Checklist

- [ ] Tests have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## What kind of change does this PR introduce?

_Refactor_

## What is the current behavior? What is the new behavior?

Test harness will fail after 3sec if prism instance has not been started correctly.

## Does this PR introduce a breaking change?

No

### CODE REVIEW

File: test-harness/index.ts [hunk 1]
```diff
@@ -43,60 +45,79 @@ describe('harness', () => {
...
+    prismMockProcessHandle.stdout.pipe(split2()).on('data', (line: string) => {
+      if (line.includes(WAIT_FOR_LINE)) {
+        clearTimeout(timeout);
+        resolve(prismMockProcessHandle);
+      }
+    });
+  });
+}
+
+function shutdownPrism(processHandle: ChildProcess, done: (...args: any[]) => any) {
```
Comment (XVincentX) [inline test-harness/index.ts]:
```suggestion
function shutdownPrism(processHandle: ChildProcess, done: jest.DoneCallback) {
```

File: test-harness/index.ts [hunk 2]
```diff
@@ -43,60 +45,79 @@ describe('harness', () => {
 
     const testText = `${file}${os.EOL}${parsed.test}`;
 
-    test(testText, done => {
+    test(testText, async done => {
```
Comment (XVincentX) [inline test-harness/index.ts]: In general mixing `async` and `done` callbacks together is not a good idea. We should either use one or the other system. In this case, let's stick with the `done` callback.

File: test-harness/index.ts [hunk 3]
```diff
@@ -43,60 +45,79 @@ describe('harness', () => {
...
-            windowsVerbatimArguments: false,
-          });
-          const output: any = parseResponse(clientCommandHandle.stdout.trim());
-          const expected: any = parseResponse((parsed.expect || parsed.expectLoose).trim());
 
-          const isXml = xmlValidator.test(
-            get(output, ['header', 'content-type'], ''),
-            expected.body
-          );
+      const prismMockProcessHandle = await startPrism(parsed.server, tmpFileHandle.name);
```
Comment (XVincentX) [inline test-harness/index.ts]: …per the comment above, we should modify the `startPrism` function to take an additional done callback to continue the flow later.

File: test-harness/index.ts (hunk 1 above)
Comment (StefanDywersant) [reply test-harness/index.ts]: It has to stay that way to be compatible both with jest call and our call (https://github.com/stoplightio/prism/pull/825/files#diff-d26ad5d0c95e0973686e7fa034e1283bR107).

File: test-harness/index.ts (hunk 3 above)
Comment (StefanDywersant) [reply test-harness/index.ts]: Done 👍

File: test-harness/index.ts (hunk 2 above)
Comment (StefanDywersant) [reply test-harness/index.ts]: You right, I updated it to use callback only.

File: test-harness/index.ts (hunk 1 above)
Comment (XVincentX) [reply test-harness/index.ts]: Ah yeah ok, got it.

**Comment by StefanDywersant** [comment]
> 2. When running `yarn test.harness` I get an empty output instead of the expected error

I checked locally. Works fine if no changes on WAIT_LINE. If you change WAIT_LINE it throws an exception after 3sec. Note that the check checks whether logline _includes_ WAIT_LINE. So if you shorten it then it can still work.

**Comment by XVincentX** [comment]
> > 1. When running `yarn test.harness` I get an empty output instead of the expected error
> 
> I checked locally. Works fine if no changes on WAIT_LINE. If you change WAIT_LINE it throws an exception after 3sec. Note that the check checks whether logline _includes_ WAIT_LINE. So if you shorten it then it can still work.

Ah! let me try again.

**Pull request merged**